pydantic/json_schema.py

```diff
@@ -445,20 +453,20 @@ def default_schema(self, schema: core_schema.WithDefaultSchema) -> JsonSchemaVal
         json_schema = self.generate_inner(schema['schema'])
 
         if 'default' in schema:
-            default = self.encode_default(schema['default'])
+            default = schema['default']
         elif 'default_factory' in schema:
-            default = self.encode_default(schema['default_factory']())
+            default = schema['default_factory']()
         else:
             raise ValueError('`schema` has neither default nor default_factory')
 
         try:
             encoded_default = self.encode_default(default)
         except TypeError:
-            # This happens if the default value is not JSON serializable; in this case, just return the inner schema
-            # Note: We could update the '$comment' field to indicate that the default value was not JSON serializable.
-            #   This would have the upside that there would be some positive indication that the default value was not
-            #   valid, but would have the downside that it would make some assumptions about how users are using the
-            #   '$comment' field. For now, I have decided not to do this.
+            self.emit_warning(
```

Comment:
i guess this is where we were going to use a method from pydantic-core, do you still need that?

Reply:
I'm not sure I understand the question. I added `self.emit_warning` to make it easier to override the warning behaviors, between that and `ignored_warning_kinds` I might have gone overboard though.

Reply:
I added a TODO: comment about this inside `self.encode_default`.